src/Simplex/Chat.hs

```diff
@@ -6225,20 +6297,42 @@ agentXFTPDeleteRcvFile aFileId fileId = do
   withAgent (`xftpDeleteRcvFile` aFileId)
   withStore' $ \db -> setRcvFTAgentDeleted db fileId
 
+agentXFTPDeleteRcvFiles :: ChatMonad m => [(XFTPRcvFile, FileTransferId)] -> m ()
+agentXFTPDeleteRcvFiles rcvFiles = do
+  let rcvFiles' = filter (not . agentRcvFileDeleted . fst) rcvFiles
+  fileIds' <- forM rcvFiles' $ \case
+    (XFTPRcvFile {agentRcvFileId = Just (AgentRcvFileId aFileId)}, fileId) -> do
+      withAgent (`xftpDeleteRcvFile` aFileId)
```

Comment:
Good candidate for agent-side batching. Alas, no batching in FileTransfer.Agent yet.